packages/addon-dev/sample-typescript-rollup.config.js

```diff
@@ -0,0 +1,113 @@
+/**
+  *  This file should live at ./config/rollup.config.js
+  *  and the babel config at ./config/babel.config.js (if the babel config is ESM)
+  *    _if the babel config is CJS, it can be at the addon root_
+  *
+  *  The command to build:
+  *
+  *    `rollup -c ./config/rollup.config.js`
+  */
+import path from 'path';
+
+import alias from '@rollup/plugin-alias';
+import multiInput from 'rollup-plugin-multi-input';
+import ts from 'rollup-plugin-ts';
+import { defineConfig } from 'rollup';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import { Addon } from '@embroider/addon-dev/rollup';
+
+import babelConfig from './babel.config';
+import packageJson from '../package.json';
+
+const addon = new Addon({
+  srcDir: 'src',
+  destDir: 'dist',
+});
+
+const transpilation = [
+  // Instruct rollup how to resolve ts and hbs imports
+  // (importing a template-only component, for example)
+  nodeResolve({ resolveOnly: ['./'], extensions: ['.js', '.ts', '.hbs'] }),
+
+  // Allow top-level imports (what folks are used to from v1 addons)
+  // During the build, anything referencing a top-level import will be
+  // replaced with a relative import.
+  // DANGER: it's somewhat easy to cause circular references with this tool
+  alias({
```

Comment:
In my opinion, having `alias` is not a requirement for typescript, so I think this example should be very similar to what the default example already has.

Reply:
I don't think we'll need alias for things like self-imports as soon as typescript lands their own support for package.json exports: https://www.typescriptlang.org/docs/handbook/esm-node.html

Reply:
oh, well if everyone is just going to support this out of the box:
https://nodejs.org/api/packages.html#packages_self_referencing_a_package_using_its_name

that's fantastic! no need for transform!